### PR TITLE
Supporting llama int4 inference using AutoGPTQ in HPU (#166)

### PR DESCRIPTION
**Pull request by HolyFalafel**
Added support for AutoGPTQ when loading quantized model, and running inference in HPU.
This will be available in v1.17

### CODE REVIEW

**Comment by emascarenhas** [comment]
Please sync your PR with main/upstream and fix any merge conflicts. Thank you.

**Comment by yafshar** [comment]
@HolyFalafel, please sync this PR with main and ping me to wrap it up. Thanks

File: examples/text-generation/README.md [hunk 1]
```diff
@@ -494,6 +494,30 @@ python ../gaudi_spawn.py --use_deepspeed --world_size 8 run_generation.py \
 
 For more details see [documentation](https://docs.habana.ai/en/latest/PyTorch/Model_Optimization_PyTorch/Optimization_in_PyTorch_Models.html#using-fused-sdpa).
 
+### Running with UINT4 using AutoGPTQ
+
+
+Llama2-7b in UINT4 is enabled using [AutoGPTQ Fork](https://github.com/HabanaAI/AutoGPTQ), which provides quantization capabilities in PyTorch.
+Currently, the support is for UINT4 inference of pre-quantized models only.
```
Comment (yafshar) [inline examples/text-generation/README.md:501]: @HolyFalafel please add the AutoGPTQ installation here,

`BUILD_CUDA_EXT=0 pip install auto-gptq --no-build-isolation`

**Comment by yafshar** [comment]
@HolyFalafel , we have the similar functionality already added. Please check https://github.com/huggingface/optimum-habana/pull/1165

**Comment by HolyFalafel** [comment]
https://github.com/huggingface/optimum-habana/pull/1364 replaces this PR

**Pull request closed without merging**